libs/core/src/components/sage-input/test/sage-input.e2e.ts

```diff
@@ -0,0 +1,52 @@
+
+import { newE2EPage } from "@stencil/core/testing";
+
+describe('sage-input', () => {
+  it('renders toggle of disabled state', async () => {
+    const page = await newE2EPage();
+    await page.setContent('<sage-input></sage-input>');
+    const component = await page.find('sage-input');
+    expect(component).toHaveClass('hydrated');
+
+    const element = await page.find('sage-input >>> input');
+    let value = await element.getProperty('disabled');
+    expect(value).toBe(false);
+
+    component.setProperty('disabled', 'true');
+    await page.waitForChanges();
+    value = await element.getProperty('disabled');
+    expect(value).toBe(true);
+  });
+
+  it('toggles an error state', async () => {
+    const page = await newE2EPage();
+
+    await page.setContent('<sage-input error-text="This is error text"></sage-input>');
+    const component = await page.find('sage-input');
+    expect(component).toHaveClass('hydrated');
+
+    const element = await page.find('sage-input >>> .sage-input__error-text');
+    expect(element.textContent).toEqual(`This is error text`);
+  });
+
+  it('renders a value onInput', async () => {
+    const page = await newE2EPage();
+
+    await page.setContent('<sage-input></sage-input>');
+    const input = await page.find('sage-input >>> input');
+
+    let value = await input.getProperty('value');
+    expect(value).toBe('');
+
+
+    await input.focus();
+    await page.waitForChanges();
+
+    const inputSpy = await page.spyOnEvent('onInput');
```

Comment:
This spec is intended to cover the lines 81 - 85 in libs/core/src/components/sage-input/sage-input.tsx, but with no success. Looking for guidance on this test to achieve that.

Reply:
Add this to `sage-input-spec.tsx`

```
it('updates value prop on value change', async () => {
    const page = await newSpecPage({
      components: [SageInput],
      html: `<sage-input value="yada-yada" />`,
    })
    const sageInput = page.root
    expect(sageInput.value).toBe('yada-yada')

    const input = sageInput.shadowRoot.querySelector('input')
    expect(input.value).toBe('yada-yada')

    input.value = 'yoda-yoda'
    input.dispatchEvent(new Event('input'))
    await page.waitForChanges()

    expect(input?.value).toEqual('yoda-yoda')

    input.value = '';
    input.dispatchEvent(new Event('input'));
    await page.waitForChanges();

    expect(input?.value).toEqual('');
  })